### PR TITLE
Refactor: move completion functions to autoload

### PR DESCRIPTION
**Pull request by stsewd**
Didn't see any improvement in startup, but if felt weird having those functions in there.

### CODE REVIEW

**Review by vigoux — APPROVED** [review]
Good idea, and I think it will be lighter in the performance too.

File: lua/nvim-treesitter/configs.lua [hunk 1]
```diff
@@ -251,29 +251,29 @@ M.commands = {
...
+      "-complete=customlist,v:lua.ts_available_modules",
+    },
   },
   TSDisableAll = {
     run = disable_all,
     args = {
       "-nargs=+",
-      "-complete=custom,v:lua.ts_available_modules"
-    }
+      "-complete=customlist,v:lua.ts_available_modules",
```
Comment (kyazdani42) [inline lua/nvim-treesitter/configs.lua:275]: oups

File: lua/nvim-treesitter/configs.lua [hunk 2]
```diff
@@ -251,29 +251,29 @@ M.commands = {
...
+      "-complete=customlist,nvim_treesitter#available_modules",
+    },
   },
   TSEnableAll = {
     run = enable_all,
     args = {
       "-nargs=+",
-      "-complete=custom,v:lua.ts_available_modules"
-    }
+      "-complete=customlist,v:lua.ts_available_modules",
```
Comment (kyazdani42) [inline lua/nvim-treesitter/configs.lua:268]: oups

**Review by kyazdani42 — COMMENTED** [review]
these two completes are not using the function !

File: lua/nvim-treesitter/configs.lua (hunk 2 above)
Comment (stsewd) [reply lua/nvim-treesitter/configs.lua:268]: sorry! diffs messed with me :(

**Pull request merged**